image/roxctl/rhtap.Dockerfile

```diff
@@ -0,0 +1,61 @@
+# openshift-golang-builder is the only way to get more recent Go version than the official ubi8/go-toolset provides.
+# See https://issues.redhat.com/browse/RHELPLAN-167618
+# Using that has few known issues:
+# - https://issues.redhat.com/browse/RHTAPBUGS-864 - deprecated-base-image-check behaves incorrectly.
+# - https://issues.redhat.com/browse/RHTAPBUGS-865 - openshift-golang-builder is not considered to be a valid base image.
+#
+# Also, we can't pin image tag or digest because currently there's no mechanism to auto-update that.
+# See https://issues.redhat.com/browse/STONEBLD-1823
+# We're targeting a floating tag here which should be reasonably safe to do as both RHEL major 8 and Go major.minor 1.20 should provide enough stability.
+FROM brew.registry.redhat.io/rh-osbs/openshift-golang-builder:rhel_8_1.20 as builder
+
+WORKDIR /go/src/github.com/stackrox/rox/app
+
+COPY . .
+
+RUN git config --global --add safe.directory /go/src/github.com/stackrox/rox/app && \
+    # TODO: this prevents hermetic builds
+    # See related: https://redhat-internal.slack.com/archives/C04PZ7H0VA8/p1696941513601719 and https://github.com/redhat-appstudio/build-definitions/pull/615
+    git fetch --tags --force && \
+    mkdir -p image/bin
+
+ENV CI=1 GOFLAGS="" GOTAGS="release"
```

Comment:
@msugakov Should we always set `GOTAGS` to "release"?

(This was copied from the midstream Dockerfile)

Reply:
Good question. `$GOTAGS` are passed to `go` command via `-tags` option (https://github.com/stackrox/stackrox/blob/main/scripts/go-tool.sh#L68-L68) which then is used to select code to compile, e.g. [mutex_dev.go](https://github.com/stackrox/stackrox/blob/main/pkg/sync/mutex_dev.go) vs [mutex_release.go](https://github.com/stackrox/stackrox/blob/main/pkg/sync/mutex_release.go).

For now, we can focus on release-like builds and keep `GOTAGS="release"` but if we look to enable developer workflow with RHTAP, we should make that flexible and follow what we do in GHA. I created https://issues.redhat.com/browse/ROX-20240 for this.

Interestingly, ROX-20240 seems to be not on a critical path for faster releases so we may want to introduce a separate bucket for it.

For now, I'd suggest to add a TODO comment:

```suggestion
# TODO(ROX-20240): enable non-release development builds.
ENV CI=1 GOFLAGS="" GOTAGS="release"
```